packages/input/src/FzCurrencyInput.vue

```diff
@@ -3,19 +3,26 @@
 </template>
 
 <script setup lang="ts">
-import { onMounted, ref } from "vue";
+import { computed, onMounted, ref } from "vue";
 import FzInput from "./FzInput.vue";
 import { FzCurrencyInputProps } from "./types";
 import { useCurrency } from "@fiscozen/composables";
 
-const fzInputRef = ref();
+const fzInputRef = ref<InstanceType<typeof FzInput>>();
+const containerRef = computed(() => fzInputRef.value?.containerRef);
+const inputRef = computed(() => fzInputRef.value?.inputRef);
 const props = defineProps<FzCurrencyInputProps>();
-const { inputRef } = useCurrency();
+const { inputRef: currencyInputRef } = useCurrency();
 
 defineEmits(["update:amount"]);
 
 onMounted(() => {
-  inputRef.value = fzInputRef.value.inputRef;
+  currencyInputRef.value = inputRef.value;
```

Comment:
are we sure we don't need to watch for inputRef changes?

Reply:
what for?

Reply:
I'm not sure what happens if the user for example puts the `FzCurrencyInput` in a v-if block and toggles it, or if it is inside a v-for with index based keys and makes modification to the list... and so on.
This runs once, are we sure that we don't want the ref but the value?

Reply:
why should this be an issue related to my changes though?

Reply:
in both cases we are assigning to the input's ref on the component's mount, it's one-off read, so nothing really changed